docs/algorithms/sig/sphincs.yml

```diff
@@ -56,9 +56,6 @@ parameter-sets:
     - architecture: x86_64
       required_flags:
       - avx2
-      operating_systems:
```

Comment:
I think we do want to document the restriction to Linux and Darwin: see [here](https://github.com/open-quantum-safe/liboqs/pull/1478).

Reply:
With the latest commit this is now as desired (spec'd by the patch). "Reason": I ran `copy_from_upstream` with parameter `-k`: This way, the patches do not get thrown away as per [this code](https://github.com/open-quantum-safe/liboqs/blob/bac716cc91c53529e3fad88e92ff9dfd19c04577/scripts/copy_from_upstream/copy_from_upstream.py#L614-L615). Again, no clue why this `rmtree` statement exists... Unless I hear objections from @bhess I'll probably simply remove this line when resolving #1584 to avoid future repeats of this issue (documentation patches simply getting nixed).

So this PR is now ready for review @SWilson4 @praveksharma .

Reply:
@baentsch the `rmtree` was originally at the end of the `def copy_from_upstream` function with the intend to delete the "repos" directory if one doesn't want to keep it. 
The `update_upstream_alg_docs.do_it(os.environ['LIBOQS_DIR'])` seems to have been added later mistakenly after removing the repos.
I'd just fix this by moving `rmtree` again to the end of the function, otherwise the -k switch doesn't have any effect.